### PR TITLE
Fix report scheduling

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for features or enhancements / 3.2 for bug fixes 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
This PR fix two things.

1, We've had duplicate report scheduling for some reports. I don't know why it happend and when, but I've added delete all report scheduler during the remove https://github.com/mautic/mautic/compare/features...kuzmany:fix-report-scheduling?expand=1#diff-e00f1c22f675d8bdab6109a17a28b8df71c282384da5f5a434e6be41362ccb59R66 

image

2. Other issue was with different timezone for reschedule. 
If you create scheduling It was scheduled to timezone use in Mautic from Configuration
If report was re-scheduled by command, then timezone was UTC
This PR should fixed it, schedule and re-schedule is still in timezone from Mautic configuration

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Set Europe/Paris timezone in Mautic Configuration
3. Create report and create scheduling every day
4. Check the reports_schedulers table in database
5. Your report is scheduled to next day 23:00:00
6. You can change the date with -1 day directy in database and run `php bin/console mautic:reports:schedule`
7. Report is send and re-schedule to next day to 00:00:00 (we expect 23:00:00) same like when we set it from Mautic
8. This PR fixed it.

### CODE REVIEW

**Comment by dennisameling** [comment]
Hi @kuzmany! Thanks for your PR 🚀 we just upgraded Mautic to Symfony 4 and this PR will need to be updated. Since this PR has conflicts now, could you please do a `git merge upstream/features` locally to resolve the conflicts? Thanks in advance!

File: app/bundles/ReportBundle/Scheduler/Model/SchedulerPlanner.php [hunk 1]
```diff
@@ -63,12 +63,15 @@ private function planScheduler(Report $report)
 
     private function removeSchedulerOfReport(Report $report)
     {
-        $scheduler = $this->schedulerRepository->getSchedulerByReport($report);
-        if (!$scheduler) {
+        $schedulers = $this->schedulerRepository->findBy(['report' => $report]);
```
Comment (escopecz) [inline app/bundles/ReportBundle/Scheduler/Model/SchedulerPlanner.php:66]: Can you marek the `getSchedulerByReport` method as deprecated as it's not used from anywhere else?

**Review by escopecz — CHANGES_REQUESTED** [review]
The code looks good. Just one small request.

File: app/bundles/ReportBundle/Scheduler/Model/SchedulerPlanner.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/ReportBundle/Scheduler/Model/SchedulerPlanner.php:66]: Done

**Comment by kuzmany** [comment]
This should be fixed also by https://github.com/mautic/mautic/pull/10322 
Then close as duplicate

> schedule time and timezone of the report to be sent

**Pull request closed without merging**